### PR TITLE
FIX Javascript, commonlib and quality tests. (ginkgo-rg)

### PR DESCRIPTION
**Pull request by hetmantsev**
**Description:** 
Fix Jasmine version (cherry-pick from ginkgo.master https://github.com/edx/edx-platform/commit/b73ebfad9ccc318f92fb0a09e3333c2bfd49def0). 

Fix date for test enrollment. https://jenkins.raccoongang.com/view/ginkgo-rg-pr/job/javascript-ginkgo-rg/9/testReport/junit/Javascript/Firefox/Course_Card_View__should_render_the_course_card_based_on_the_data_enrolled/

Fix date for arabic language. https://jenkins.raccoongang.com/view/ginkgo-rg-pr/job/javascript-ginkgo-rg/11/testReport/junit/Javascript/Firefox/DateUtilFactory__localizes_some_times/

Fix code quality.

Fix commonlib-unit tests (by @AndreyLykhoman )

Fix diff-coverage for js and quality tests (added TARGET_BRANCH variable to generic-ci test script. Previously it was hardcoded in paver.)

**Youtrack:** 
https://youtrack.raccoongang.com/issue/DOI-61

**Reviewers:**
- @dgamanenko 
- @sidorovdmitry 
- @a-kryachko 
- @AndreyLykhoman 
- @oksana-slu 

**Merge checklist:**
- [x] All reviewers approved
- [x] CI build is green
- [x] Commits are squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

File: lms/static/js/spec/dateutil_factory_spec.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ define(['../dateutil_factory.js'], function(DateUtilIterator) {
                 var testLangs = {
                     en: 'Due Oct 14, 2016 08:00 UTC',
                     ru: 'Due 14 окт. 2016 г. 08:00 UTC',
-                    ar: 'Due ١٤ تشرين الأول أكتوبر ٢٠١٦ ٠٨:٠٠ UTC',
+                    ar: 'Due ١٤ أكتوبر ٢٠١٦ ٠٨:٠٠ UTC',
```
Comment (sidorovdmitry) [inline lms/static/js/spec/dateutil_factory_spec.js]: What is this?

File: package.json [hunk 2]
```diff
@@ -35,7 +35,7 @@
     "eslint-config-edx": "^3.0.0",
     "eslint-config-edx-es5": "^2.0.0",
     "eslint-import-resolver-webpack": "^0.8.1",
-    "jasmine-core": "^2.6.4",
+    "jasmine-core": "2.6.4",
```
Comment (sidorovdmitry) [inline package.json]: Why?

File: lms/static/js/spec/dateutil_factory_spec.js (hunk 1 above)
Comment (hetmantsev) [reply lms/static/js/spec/dateutil_factory_spec.js]: I've rebased these commits and added more info in description.

File: package.json (hunk 2 above)
Comment (hetmantsev) [reply package.json]: Thats cherry-pick from ginkgo.master branch.

https://github.com/edx/edx-platform/commit/b73ebfad9ccc318f92fb0a09e3333c2bfd49def0

**Comment by hetmantsev** [comment]
test this please

**Comment by hetmantsev** [comment]
start unit tests

**Comment by hetmantsev** [comment]
start unit tests

**Comment by hetmantsev** [comment]
start quality tests

**Comment by hetmantsev** [comment]
start quality tests

**Comment by hetmantsev** [comment]
start quality test

**Comment by hetmantsev** [comment]
start unit tests

**Comment by hetmantsev** [comment]
start unit tests

**Pull request merged**